### DISCUSSION TITLE
Fund. SQL Server-Inserindo progresso - Gostaria de entender porque não consegui inserir o progresso usando variaveis

### DISCUSSION THREAD

**Discussion by oimarcella** [balta-io/forum#447] [Tira Dúvidas] [answered]
Fiz um script para inserir um aluno na tabela Student, e usei uma variável para armazenar o id.
```DECLARE @idStudent UNIQUEIDENTIFIER = (SELECT NEWID())```

Só que ao tentar reutilizar essa variável para criar o vínculo do aluno com o curso, causou um erro:
```The INSERT statement conflicted with the FOREIGN KEY constraint "FK_StudentCourse_Student_StudentId". The conflict occurred in database "balta", table "dbo.Student", column 'Id'.```

**Comment by oimarcella** [comment, marked as answer]
Eu acabei tendo que usar uma string fixa na variável para funcionar, porém imaginei que com a variável seria mais "prático", se é que posso dizer assim.

Fiz da seguinte maneira:
```DECLARE @idStudent UNIQUEIDENTIFIER = 'c9098db1-9004-47f8-a548-9bde8c7b363c'```

**Reply by andrebaltieri** [reply]
Hummm, confesso que nunca usei assim, não sabia desse comportamento.
Se puder colar o script todo aqui para eu testar, ai posso tentar entender o que aconteceu por baixo dos panos.

Obrigado por compartilhar!
